src/cp_parallel/cp_parallel.py

```diff
@@ -0,0 +1,153 @@
+
+        # Create and configure a logger for this plate
+        logger = logging.getLogger(f"{run_name}_{plate_name}")
+        logger.setLevel(logging.INFO)
+
+        # Avoid adding multiple handlers to the same logger
+        if not logger.handlers:
+            fh = FileHandler(log_file_path)
+            log_format = "[%(asctime)s] [Process ID: %(process)d] %(message)s"
+            formatter = Formatter(log_format)
+            fh.setFormatter(formatter)
+            logger.addHandler(fh)
+
+        # Log the information
+        logger.info(f"Plate Name: {plate_name}")
+        logger.info(f"Output String: {output_string}")
+
+        # Explicitly close and remove the handler after logging
+        for handler in logger.handlers[:]:
+            handler.close()
+            logger.removeHandler(handler)
+
+
+def run_cellprofiler_parallel(
+    plate_info_dictionary: dict,
+    run_name: str,
+) -> None:
+    """
+    This function utilizes multi-processing to run CellProfiler pipelines in parallel.
+
+    Args:
+        plate_info_dictionary (dict): dictionary with all paths for CellProfiler to run a pipeline
+        run_name (str): a given name for the type of CellProfiler run being done on the plates (example: whole image features)
+
+    Raises:
+        FileNotFoundError: if paths to pipeline and images do not exist
+    """
+    # create a list of commands for each plate with their respective log file
+    commands = []
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
+
+    # iterate through each plate in the dictionary
+    for _, info in plate_info_dictionary.items():
+        # set paths for CellProfiler
+        path_to_pipeline = info["path_to_pipeline"]
+        path_to_images = info["path_to_images"]
+        path_to_output = info["path_to_output"]
+
+        # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+        if not pathlib.Path(path_to_pipeline).resolve(strict=True):
+            raise FileNotFoundError(
+                f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist"
+            )
+        if not pathlib.Path(path_to_images).is_dir():
+            raise FileNotFoundError(
+                f"Directory '{pathlib.Path(path_to_images).name}' does not exist or is not a directory"
+            )
+        # make output directory if it is not already created
+        pathlib.Path(path_to_output).mkdir(exist_ok=True)
+
+        # creates a command for each plate in the list
+        command = [
+            "cellprofiler",
+            "-c",
+            "-r",
+            "-p",
+            path_to_pipeline,
+            "-o",
+            path_to_output,
+            "-i",
+            path_to_images,
+        ]
+        # creates a list of commands
+        commands.append(command)
+
+    # set the number of CPUs/workers as the number of commands
+    num_processes = len(commands)
```

Comment:
But if someone has 3 commands to run based on their dictionary, for example, why would they need to start more than that number of processes?

It is an easy addition but I am not sure an instance where an individual would use that. I could just be misunderstanding too.

Reply:
I have added issue #3 in reference to this.

Reply:
> But if someone has 3 commands to run based on their dictionary, for example, why would they need to start more than that number of processes?

Yes, great point! This would I think be an edge case that could be covered by another if-condition.

My initial comments mostly pertained to times where someone would want to only provide a limited number of processors (they may have other things running). For example, if there are 5 CP runs to accomplish but the person wants to only use 3 cores to accomplish that work (each additional job beyond 3 would begin as the earlier submitted ones complete).